### PR TITLE
[ci][multipy/4] add python 3.8+3.11 test jobs for serve

### PR DESCRIPTION
**Pull request by can-anyscale**
Add python 3.8 + 3.11 test jobs for serve. Add them to continuous run.
- Unify the old 3.11 job with the multipy workflow

Test:
- CI

### CODE REVIEW

File: ci/docker/serve.build.Dockerfile [hunk 1]
```diff
@@ -3,7 +3,6 @@
 ARG DOCKER_IMAGE_BASE_BUILD=cr.ray.io/rayproject/oss-ci-base_build
 FROM $DOCKER_IMAGE_BASE_BUILD
 
-ARG PYTHON_VERSION
```
Comment (can-anyscale) [inline ci/docker/serve.build.Dockerfile:6]: inherit from the base so no need to re-declare here

**Review by aslonnie — COMMENTED** [review]
do we have proof that 3.11 tests work?

**Review by aslonnie — APPROVED** [review]

**Comment by can-anyscale** [comment]
yess, serve has been running python 3.11 tests in CI for a while; this PR just adds 3.8 and migrate the existing 3.11 jobs to the new wanda config

File: ci/docker/serve.build.Dockerfile (hunk 1 above)
Comment (aslonnie) [reply ci/docker/serve.build.Dockerfile:6]: I do not think docker build inherit build args.

if we are dropping this, we should just remove line 24-29

Comment (can-anyscale) [reply ci/docker/serve.build.Dockerfile:6]: w00t nice catch

**Review by GeneDer — APPROVED** [review]
LGTM :)

**Review by edoakes — APPROVED** [review]

**Pull request merged**